src/Microsoft.AspNetCore.Mvc.Core/Internal/TaskCache.cs

```diff
@@ -8,22 +8,12 @@ namespace Microsoft.AspNetCore.Mvc.Internal
     public static class TaskCache
     {
 #if NET451
-        static readonly Task _completedTask = Task.FromResult(0);
-#endif
-
-        /// <summary>Gets a task that's already been completed successfully.</summary>
-        /// <remarks>May not always return the same instance.</remarks>        
-        public static Task CompletedTask
-        {
-            get
-            {
-#if NET451
-                return _completedTask;
+        /// <summary>Gets a task that's already been completed successfully.</summary> 
+        public static readonly Task CompletedTask = Task.FromResult(0);
 #else
-                return Task.CompletedTask;
```

Comment:
This was actually showing up in the profile. It's not getting inlined, and the implementation of `Task.CompletedTask` does a volatile read.